### PR TITLE
Update to tools/ubuntu-setup to support xenial and later Ubuntu releases

### PR DESCRIPTION
**Pull request by jonpspri**
Updated to tools/ubuntu-setup/*.sh to support xenial and later versions of Ubuntu as part of the install process.  Trusty support remains.  Java uses OpenJDK 8 in Xenial and Docker uses latest stable Docker CE.  Support for Xenial+ is experimental and the scripts print appopriate warnings.

### CODE REVIEW

File: tools/ubuntu-setup/all.sh [hunk 1]
```diff
@@ -10,15 +18,35 @@ echo "*** installing basics"
...
-echo "*** installing java"
-"$SCRIPTDIR/java8.sh"
-
-echo "*** install scala"
-"$SCRIPTDIR/scala.sh"
+u_release="$(lsb_release -rs)"
 
-echo "*** installing docker"
-"$SCRIPTDIR/docker.sh"
+echo "*** installing java"
```
Comment (csantanapr) [inline tools/ubuntu-setup/all.sh]: No need for this, let's install openjdk for any linux ubuntu 14 or 16

Comment (jonpspri) [reply tools/ubuntu-setup/all.sh]: Unfortunately, OpenJDK 8 is not available in trusty without jumping through the same hoops as 'java.sh' already does.

Comment (csantanapr) [reply tools/ubuntu-setup/all.sh]: Could you modify java.sh to pump and install OpenJDK ?

**Comment by jonpspri** [comment]
@csantanapr Please look at latest.  OpenJDK 8 seems to be installing for everything now.

File: tools/ubuntu-setup/java8.sh [hunk 2]
```diff
@@ -2,8 +2,10 @@
 set -e
 set -x
 
-sudo apt-get install -y software-properties-common
-sudo add-apt-repository -y ppa:webupd8team/java
-sudo apt-get update -y
-echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | sudo /usr/bin/debconf-set-selections
-sudo apt-get install -y oracle-java8-installer
+if [ "$(lsb_release -cs)" == "trusty" ]; then
```
Comment (csantanapr) [inline tools/ubuntu-setup/java8.sh]: Nice to have this consistency of openjdk8 for both ubuntu 14 or above

File: tools/ubuntu-setup/all.sh [hunk 3]
```diff
@@ -1,4 +1,12 @@
 #!/bin/bash
+
+#  This script can be tested for validity by doing something like:
+#
+#  docker run -v "${OPENWHISK_HOME}:/openwhisk" ubuntu:trusty \
+#    sh -c 'apt-get update && apt-get -y install sudo && /openwhisk/tools/ubuntu-setup/all.sh'
```
Comment (csantanapr) [inline tools/ubuntu-setup/all.sh]: So good you added this comment

Comment (jonpspri) [reply tools/ubuntu-setup/all.sh]: Yeah.  It took me a little time to line everything up, but it makes for many fewer breaks in production.

File: tools/ubuntu-setup/java8.sh (hunk 2 above)
Comment (jonpspri) [reply tools/ubuntu-setup/java8.sh]: Just hoping 'jonathonf' keeps his PPA up-to-date until trusty goes EOL.

**Comment by csantanapr** [comment]
@jonpspri I got the following error specifying `BOX = "ubuntu/xenial64"`
```
👽  $ ./hello
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'ubuntu/xenial64' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: >= 0
The box 'ubuntu/xenial64' could not be found or
could not be accessed in the remote catalog. If this is a private
box on HashiCorp's Atlas, please verify you're logged in via
`vagrant login`. Also, please double-check the name. The expanded
URL and error message are shown below:

URL: ["https://atlas.hashicorp.com/ubuntu/xenial64"]
Error: The requested URL returned error: 404 Not Found
```

**Comment by csantanapr** [comment]
Upgrading to virtualbox 5.2.6 and vagrant 2.0.2
Will report results

**Comment by csantanapr** [comment]
It worked after updating @jonpspri 
```
    default: ok: whisk auth set. Run 'wsk property get --auth' to see the new value.
    default: ok: whisk API host set to 192.168.33.16
    default: Swagger UI URL: https://192.168.33.16/api/v1/docs/index.html?url=/api/v1/api-docs
    default: ++ echo 'Swagger UI URL: https://192.168.33.16/api/v1/docs/index.html?url=/api/v1/api-docs'
    default: ++ wsk action invoke /whisk.system/utils/echo -p message hello --result
    default: {
    default:     "message": "hello"
    default: }
    default: +++ date
    default: ++ echo 'Wed Feb 28 20:50:55 UTC 2018: build-deploy-end'
```

**Pull request merged**